sru/release-27.14/test-migrate-user-config.sh

```diff
@@ -0,0 +1,273 @@
+    echo
+    lxc exec $name -- pro config show
+    echo "-------------------------------------------"
+
+    lxc delete --force $name
+    echo "###########################################"
+}
+
+function test_migration_failure {
+    series=$1
+    old_version=$2
+    echo -e "\n\n###########################################"
+    echo "## $series: $old_version -> 27.14: migration failure"
+    echo "###########################################"
+    name=$(echo $series-$old_version | tr .~ -)
+
+    echo -e "\n-------------------------------------------"
+    echo "** launching container"
+    echo "-------------------------------------------"
+    lxc launch -q ubuntu-daily:$series $name
+    sleep 5
+    lxc exec $name -- apt-get update > /dev/null
+    echo "-------------------------------------------"
+
+    install_old_version $name $series $old_version
+
+    echo -e "\n-------------------------------------------"
+    echo "** mess up uaclient.conf to be invalid yaml"
+    echo "-------------------------------------------"
+    lxc exec $name -- sh -c "echo {{{ >> /etc/ubuntu-advantage/uaclient.conf"
+    echo "-------------------------------------------"
+    echo -e "\n-------------------------------------------"
+    echo "** Show uaclient.conf"
+    echo "-------------------------------------------"
+    lxc exec $name -- cat /etc/ubuntu-advantage/uaclient.conf
+    echo "-------------------------------------------"
+
+    upgrade_to_proposed $name "WARNING MESSAGE PRESENT AND NO CONFFILE PROMPT"
+
+    echo -e "\n-------------------------------------------"
+    echo "** Backup file is gone"
+    echo "-------------------------------------------"
+    lxc exec $name -- ls -la /etc/ubuntu-advantage/
+    echo "-------------------------------------------"
+    echo -e "\n-------------------------------------------"
+    echo "** Show uaclient.conf - backup was restored"
+    echo "-------------------------------------------"
+    lxc exec $name -- cat /etc/ubuntu-advantage/uaclient.conf
+    echo "-------------------------------------------"
+    echo -e "\n-------------------------------------------"
+    echo "** Show user config"
+    echo "-------------------------------------------"
+    lxc exec $name -- sh -c "ls -al /var/lib/ubuntu-advantage/user-config.json || true"
+    echo "-------------------------------------------"
+
+    lxc delete --force $name
+    echo "###########################################"
+}
+
+# xenial
+test_normal_upgrade                 xenial 27.11.3~16.04.1
+test_normal_upgrade                 xenial 27.13.1~16.04.1
+test_apt_news_false_upgrade         xenial 27.11.3~16.04.1
+test_apt_news_false_upgrade         xenial 27.13.1~16.04.1
+test_uaclient_conf_changes_upgrade  xenial 27.11.3~16.04.1
+test_uaclient_conf_changes_upgrade  xenial 27.13.1~16.04.1
+test_migration_failure              xenial 27.11.3~16.04.1
+test_migration_failure              xenial 27.13.1~16.04.1
```

Comment:
@orndorffgrant maybe I am missing something, but I think we are still missing the check for the `uaclient.conf.preinst-backup` file

Reply:
Ah oops yep I was focused on everything else and just forgot - I'll add it

Reply:
Oh wait sorry I did add it. There are now sections called "Backup file is gone after successful migration" where we `ls -al /etc/ubuntu-advantage`

Or did you want something more?